### PR TITLE
[move-mutator][move-spec-test] milestone review remarks implementation and cleanup

### PR DESCRIPTION
**Pull request by asmie**
### Description

That's a technical PR which implements:
- milestone review remarks;
- comments and log messages cleanups;
- code shortenings and small fixes;
- test fixes to run with more than one thread;
- running Move Prover correctly when called from `aptos` command;
- clippy cleanups;
- minor refactoring like changing variable or struct names.

### CODE REVIEW

File: third_party/move/tools/move-mutator/doc/design.md [hunk 1]
```diff
@@ -160,6 +160,19 @@ Sample configuration file:
...
+```toml
+[project]
+move_sources = ["/path/to/move/source"]
+[mutation]
+operators = ["operator1", "operator2"]
+categories = ["category1", "category2"]
+[[individual]]
+file = "/path/to/file"
+verify_mutants = true
+include_functions = ["function1", "function2"]
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/doc/design.md:172]: Is this a fully qualified function name? Eg, address + module + function name?

File: third_party/move/tools/move-mutator/src/mutate.rs [hunk 2]
```diff
@@ -114,56 +113,56 @@ fn parse_expressions(exp: Vec<parser::ast::Exp>) -> anyhow::Result<Vec<Mutant>>
...
+        ast::Exp_::IfElse(exp1, exp2, exp3) => {
             let mut mutants = parse_expression_and_find_mutants(*exp1)?;
             mutants.extend(parse_expression_and_find_mutants(*exp2)?);
             if let Some(exp3) = exp3 {
                 mutants.extend(parse_expression_and_find_mutants(*exp3)?);
             }
             Ok(mutants)
         },
-        parser::ast::Exp_::Quant(_, _, vexp, lexp, exp) => {
+        ast::Exp_::Quant(_, _, vexp, lexp, exp) => {
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/mutate.rs:165]: Isn't this a specification language only construct? Why would we want to to mutate it?

**Review by vineethk — COMMENTED** [review]
lgtm, but have a comment about what gets mutated.

File: third_party/move/tools/move-mutator/doc/design.md (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/doc/design.md:172]: No, that's only name of the function.

It's just like for the modules - every module that equals will be handled and the same is here - every function that fits the name here will be mutated.

File: third_party/move/tools/move-mutator/src/mutate.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/mutate.rs:165]: Yes, indeed. I've noticed it a bit later, when I was transforming AST from 'parser' to 'typing' stage.
It's removed in the next PRs.

File: third_party/move/tools/move-mutator/doc/design.md (hunk 1 above)
Comment (vineethk) [reply third_party/move/tools/move-mutator/doc/design.md:172]: But how do you mutate a function `foo` in a given module, and not all the `foo` functions in different modules? Function names can clash between different modules.

Comment (asmie) [reply third_party/move/tools/move-mutator/doc/design.md:172]: Of course. If user would like to narrow functions to only one module then user can use a field that allows to choose only certain modules.

You're right, that field was missing in the sample doc. 
Doc has been updated in #86.

**Pull request merged**